### PR TITLE
create pending pipeline

### PR DESCRIPTION
**Pull request by LyndseyBu**
# Changes
as described in : https://github.com/tektoncd/dashboard/issues/2114

Allow for the creation of a pending PipelineRun
image
image

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/main/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/main/standards.md#principles) (if user facing)
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/main/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/dashboard/blob/main/CONTRIBUTING.md)
for more details._

### CODE REVIEW

**Comment by LyndseyBu** [comment]
@AlanGreene ,  unsure about some of the strings here, and do we need to translate labels and aria-labels?

File: src/containers/CreatePipelineRun/CreatePipelineRun.js [hunk 1]
```diff
@@ -57,7 +58,8 @@ const initialState = {
   submitError: '',
   timeout: '60',
   validationError: false,
-  validTimeout: true
+  validTimeout: true,
+  pendingPipelineStatus: ''
```
Comment (AlanGreene) [inline src/containers/CreatePipelineRun/CreatePipelineRun.js:62]: nit: put this above `pipelineError` so they're in alphabetical order, makes it easier to find things in the list and avoid duplicates etc.

File: src/containers/CreatePipelineRun/CreatePipelineRun.js [hunk 2]
```diff
@@ -105,7 +107,8 @@ function CreatePipelineRun(props) {
       submitError,
       timeout,
       validationError,
-      validTimeout
+      validTimeout,
+      pipelinePendingStatus
```
Comment (AlanGreene) [inline src/containers/CreatePipelineRun/CreatePipelineRun.js:111]: nit: same as above, list these alphabetically

File: src/containers/CreatePipelineRun/CreatePipelineRun.js [hunk 3]
```diff
@@ -667,6 +682,18 @@ function CreatePipelineRun(props) {
...
             }
           />
+          <Toggle
+            defaultToggled={false}
+            id="pending-pipeline-toggle"
+            labelText={intl.formatMessage({
+              id: 'dashboard.createPipelineRun.pendingPipilinRun',
+              defaultMessage: 'Create PipelineRun in pending state'
+            })}
+            onChange={handleToggleChange}
```
Comment (AlanGreene) [inline src/containers/CreatePipelineRun/CreatePipelineRun.js:692]: `onToggle` seems to give a cleaner interface here, providing the value of `checked` as the first param so we don't have to deal with the event object at all.

File: src/containers/CreatePipelineRun/CreatePipelineRun.js [hunk 4]
```diff
@@ -667,6 +682,18 @@ function CreatePipelineRun(props) {
               setState(state => ({ ...state, timeout: value }))
             }
           />
+          <Toggle
+            defaultToggled={false}
+            id="pending-pipeline-toggle"
+            labelText={intl.formatMessage({
+              id: 'dashboard.createPipelineRun.pendingPipilinRun',
```
Comment (AlanGreene) [inline src/containers/CreatePipelineRun/CreatePipelineRun.js:689]: typo: should be `.pendingPipelineRun`, we could actually shorten this to just `.pending` since it's already scoped to the Create PipelineRun page, or maybe `.status.pending` for future expansion.

File: src/containers/CreatePipelineRun/CreatePipelineRun.js [hunk 5]
```diff
@@ -667,6 +682,18 @@ function CreatePipelineRun(props) {
...
+              defaultMessage: 'Create PipelineRun in pending state'
+            })}
+            onChange={handleToggleChange}
+            labelA="disabled"
+            labelB="enabled"
+            aria-label="Pending PipelineRun state"
```
Comment (AlanGreene) [inline src/containers/CreatePipelineRun/CreatePipelineRun.js:695]: These should all be translated, and let's go with 'Enabled' and 'Disabled' for consistency with other labels in the UI. I spotted a few places where our aria-labels aren't currently translated but we can do a separate scrub of those.

Do we actually need to provide an `aria-label` here or is `labelText` sufficient?

File: src/containers/CreatePipelineRun/CreatePipelineRun.js [hunk 6]
```diff
@@ -134,6 +137,17 @@ function CreatePipelineRun(props) {
...
+  function handleToggleChange(event) {
+    const isPending = event.target.checked;
+    if (isPending) {
+      setState(state => ({
+        ...state,
+        pipelinePendingStatus: 'PipelineRunPending'
+      }));
+    } else {
+      setState(state => ({ ...state, pipelinePendingStatus: '' }));
+    }
```
Comment (AlanGreene) [inline src/containers/CreatePipelineRun/CreatePipelineRun.js:149]: we can flip this inside out since the only piece changing is the status value:

```js
setState(state => ({
  ...state,
  pipelinePendingStatus: isPending ? 'PipelineRunPending' : ''
}));
```

**Review by AlanGreene — CHANGES_REQUESTED** [review]
Looks good, thanks @LyndseyBu. Just a few minor nits and suggestions. Also answered your question about translations.

File: src/containers/CreatePipelineRun/CreatePipelineRun.js (hunk 5 above)
Comment (LyndseyBu) [reply src/containers/CreatePipelineRun/CreatePipelineRun.js]: I removed the `aria-label `, I was unsure if both we needed, some of the carbon examples have both and others just have labeText, so we I guess we're fine to go to with that

**Review by AlanGreene — APPROVED** [review]
Looks good @LyndseyBu, thanks. Just needs a rebase then it's good to go.
